MLS.Agent.Tests/XplotKernelExtensionTests.cs

```diff
@@ -0,0 +1,98 @@
+        {
+        }
+
+        [Fact]
+        public async Task When_a_chart_is_returned_the_value_produced_has_html_with_the_require_config_call()
+        {
+            var kernel = CreateKernel();
+            kernel.UseDefaultExtensions();
+
+            await kernel.SendAsync(new SubmitCode("using XPlot.Plotly;"));
+            await kernel.SendAsync(new SubmitCode("new PlotlyChart()"));
+
+            KernelEvents
+                .ValuesOnly()
+                .OfType<ValueProduced>()
+                .Should().
+                ContainSingle(valueProduced =>
+                    valueProduced.FormattedValues.Any(formattedValue =>
+                       formattedValue.Value.ToString().Contains("require.config({paths:{plotly:\'https://cdn.plot.ly/plotly-latest.min\'}});")
+                 ));
+        }
+
+        [Fact]
+        public async Task When_a_chart_is_returned_the_value_produced_hash_the_plotly_newPlot_call()
+        {
+            var kernel = CreateKernel();
+            kernel.UseDefaultExtensions();
+
+            await kernel.SendAsync(new SubmitCode("using XPlot.Plotly;"));
+            await kernel.SendAsync(new SubmitCode("new PlotlyChart()"));
+
+            KernelEvents.
+                ValuesOnly()
+                .OfType<ValueProduced>()
+                .Should().
+                ContainSingle(valueProduced =>valueProduced.FormattedValues.Any(formattedValue =>
+                         formattedValue.Value.ToString().Contains("Plotly.newPlot")
+                 ));
+        }
+
+        [Fact]
+        public async Task When_a_chart_is_returned_the_value_produced_has_the_require_plotly()
+        {
+            var kernel = CreateKernel();
+            kernel.UseDefaultExtensions();
+
+            await kernel.SendAsync(new SubmitCode("using XPlot.Plotly;"));
+            await kernel.SendAsync(new SubmitCode("new PlotlyChart()"));
+
+            KernelEvents.
+                ValuesOnly()
+                .OfType<ValueProduced>()
+                .Should().
+                ContainSingle(valueProduced =>
+                    valueProduced.FormattedValues.Any(formattedValue =>
+                        formattedValue.Value.ToString().Contains("require([\'plotly\'], function(Plotly)")
+                 ));
+        }
+
+        [Fact]
+        public async Task When_a_chart_is_returned_the_value_produced_has_the_mime_type_html()
+        {
+            var kernel = CreateKernel();
+            kernel.UseDefaultExtensions();
+
+            await kernel.SendAsync(new SubmitCode("using XPlot.Plotly;"));
+            await kernel.SendAsync(new SubmitCode("new PlotlyChart()"));
+
+            KernelEvents.ValuesOnly()
+                .OfType<ValueProduced>()
+                .Should().
+                ContainSingle(valueProduced =>
+                    valueProduced.FormattedValues.Any(formattedValue =>
+                        formattedValue.MimeType == "text/html"
```

Comment:
I think it would be more precise to test the contents and mimetype against the same formattedValue instance.